app/services/postgres_data_store.rb

```diff
@@ -18,7 +18,16 @@ def add_item(item)
   def append_entry(entry)
     entry_type = entry.entry_type.to_sym
     item = @items[entry.item_hash]
-    previous_entry_number = @records[entry_type].has_key?(entry.key) ? @records[entry_type][entry.key].last[:entry_number] : nil
+    previous_entry_number_from_memory = @records[entry_type].key?(entry.key) ? @records[entry_type][entry.key].last[:entry_number] : nil
```

Comment:
I'd be tempted to make this even simpler and say something along the lines of:

```
previous_entry_number = @records[entry_type].key?(entry.key) ? @records[entry_type][entry.key].last : Record.where(register_id: @register.id, entry_type: entry.entry_type.to_s, key: entry.key.to_s).first
```

Then you could say:
```
unless previous_entry_number.nil?
  previous_entry_number = previous_entry_number[:entry_number]
end
```

or something to that avail...